src/components/backgroundselector/BackgroundSelectorDirective.js

```diff
@@ -0,0 +1,138 @@
+(function() {
+  goog.provide('ga_backgroundselector_directive');
+
+  goog.require('ga_map');
+  goog.require('ga_permalink');
+
+  var module = angular.module('ga_backgroundselector_directive', [
+    'ga_map',
+    'ga_permalink',
+    'pascalprecht.translate'
+  ]);
+
+  module.directive('gaBackgroundSelector',
+    function($document, gaPermalink, gaLayers, gaLayerFilters,
+      gaBrowserSniffer) {
+      return {
+        restrict: 'A',
+        templateUrl:
+            'components/backgroundselector/partials/backgroundselector.html',
+        scope: {
+          map: '=gaBackgroundSelectorMap'
+        },
+        link: function(scope, elt, attrs) {
+          scope.isBackgroundSelectorClosed = true;
+          scope.embed = gaBrowserSniffer.embed;
+          var map = scope.map;
+          var firstLayerChangeEvent = true;
```

Comment:
yeah, but then currentTopic will be set, and the part guarded by `if (scope.currentTopic)` won't be executed again. Only the first time. The idea is the remove firstLayerChangeEvent variable and us currentTopic to do the same thing.